### PR TITLE
Adding check to avoid ETH transfer when not required

### PR DESCRIPTION
**Pull request by KalpitaMandal**
# Description
If a borrower calls depositCollateral for an ERC20 token but sends ETH also with the transaction, hence msg.value is greater than 0, ETH is forwarded to the SavingsAccount and is not accounted for. The following code in the _depositCollateral sends the current msg.value to SavingAccounts even if the user is depositing an amount of ERC20 tokens as collateral:

```
if (_collateralAsset == address(0)) { 

require(msg.value == _amount, "CreditLine::_depositCollateral - value to transfer doesn't match argument"); 

} else {

 IERC20(_collateralAsset).safeTransferFrom(msg.sender, address(this), _amount); 

IERC20(_collateralAsset).approve(_strategy, _amount); //audit-info approving strategy to transfer _amount from this contract 

} 

uint256 _sharesReceived = _savingsAccount.deposit{value: msg.value}(_amount, _collateralAsset, _strategy, address(this));
```

# Integrations Checklist

- [ ]  Have any function signatures changed? If yes, outline below.
- [ ]  Have any features changed or been added? If yes, outline below.
- [ ]  Have any events changed or been added? If yes, outline below.
- [ ]  Has all documentation been updated?

# Changelog

- Adding check to avoid ETH transfer when not required.

# Event Signature Changes

# Function Signature Changes

# Features

# Events

### CODE REVIEW

**Comment by prateekreddy** [comment]
@KalpitaMandal There are other places as well where this issue exists, can you please review and fix in all other functions as well

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/CreditLine/CreditLine.sol [hunk 1]
```diff
@@ -636,6 +636,9 @@ contract CreditLine is ReentrancyGuard, OwnableUpgradeable {
         bool _fromSavingsAccount
     ) internal {
         require(creditLineConstants[_id].lender != msg.sender, 'lender cant deposit collateral');
+        if(Address(creditLineConstants[_id].collateralAsset) != Address(0)) {
+            require(msg.value == 0, '_depositCollateral: ETH is not required for this operation');
```
[the remark this answers is not in the document]
Comment (KalpitaMandal) [reply contracts/CreditLine/CreditLine.sol:640]: Added this check in other places where we are expecting tokens

[the remark this answers is not in the document]
Comment (KalpitaMandal) [reply contracts/CreditLine/CreditLine.sol:640]: Added this check in functions accepting tokens

**Comment by prateekreddy** [comment]
ETH is not directly used anymore. Only WETH is used.

**Pull request closed without merging**